tcpproxy.go

```diff
@@ -93,11 +94,19 @@ func equals(want string) Matcher {
 
 // config contains the proxying state for one listener.
 type config struct {
-	routes      []route
+	sync.Mutex  // protect r/w of routes
+	nextRouteId int
+	routes      map[int]route
 	acmeTargets []Target // accumulates targets that should be probed for acme.
 	stopACME    bool     // if true, AddSNIRoute doesn't add targets to acmeTargets.
 }
 
+func NewConfig() (cfg *config) {
```

Comment:
needs docs.

and unname result parameter per https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

but actually I'd just delete this and let the zero value of Config be useful. You can initialize the routes map as needed.

Reply:
Removed the constructor.